jobrunner/local_run.py

```diff
@@ -171,30 +173,32 @@ def create_and_run_jobs(
 
     jobs = find_where(Job)
 
-    missing_docker_images = get_missing_docker_images(jobs)
-    missing_docker_images = temporary_docker_auth_workaround(missing_docker_images)
-    if missing_docker_images:
-        print("Fetching missing docker images")
-        for image in missing_docker_images:
+    for image in get_docker_images(jobs):
+        is_stata = 'stata-mp' in image
+
+        if is_stata and config.STATA_LICENSE is None:
+            config.STATA_LICENSE = get_stata_license()
+            if config.STATA_LICENSE is None:
+                # TODO switch this to failing when the stata image requires it
+                print('WARNING: no STATA_LICENSE found')
+                
+        if not docker.image_exists_locally(image):
+            print(f"Fetching missing docker image {image}")
```

Comment:
The most trivial of bikesheddy comments, but is this extra print necessary given the one below? Could maybe combine them as:
```py
print(f"\nFetching missing docker image: docker pull {image}")
```